module/item/item.js

```diff
@@ -831,13 +840,75 @@ export class PBItem extends Item {
    */
   async equip() {
     await this.setEquipped(true);
+    await this._transferEffectsToActor(true);
   }
 
   /**
    * @return {Promise<void>}
    */
   async unequip() {
     await this.setEquipped(false);
+    await this._transferEffectsToActor(false);
+  }
+
+  /**
+   * Transfer item effects to/from actor when equipped/unequipped
+   * @param {Boolean} equipped - true to add effects, false to remove them
+   * @return {Promise<void>}
+   * @private
+   */
+  async _transferEffectsToActor(equipped) {
+    if (!this.parent || !this.effects.size) return;
+
+    const actor = this.parent;
+    
+    if (equipped) {
+      // Check if effects are already applied to avoid duplicates
+      const existingEffects = actor.effects.filter(effect => 
+        effect.origin === this.uuid || 
+        effect.flags?.core?.sourceId === this.uuid
+      );
+      
+      if (existingEffects.length > 0) {
+        console.log(`Effects already applied for ${this.name}, skipping duplicate creation`);
```

Comment:
Is there a chance that changing the effects of an item would not update the effect because it gets skipped? Note I haven't had a chance to check out this branch and use it yet.